include/hse/hse.h

```diff
@@ -863,6 +863,50 @@ hse_kvs_cursor_read(
     size_t *               val_len,
     bool *                 eof);
 
+/** @brief Iteratively access the elements pointed to by the cursor.
+ *
+ * Read a key-value pair from the cursor, advancing the cursor past its current
+ * location. The key-value pair will be copied into the user's buffer(s). If the
+ * argument @p valbuf is NULL, only the key is read.
+ *
+ * @note If the cursor is at EOF, attempts to read from it will not change the
+ * state of the cursor.
+ * @note This function is thread safe across disparate cursors.
+ *
+ * <b>Flags:</b>
+ * @arg 0 - Reserved for future use.
+ *
+ * @param cursor: Cursor handle from hse_kvs_cursor_create().
+ * @param flags: Flags for operation specialization.
+ * @param[in,out] keybuf: Buffer into which the next key will be copied.
+ * @param keybuf_sz: Size of @p keybuf.
+ * @param[out] key_len: Length of the key.
+ * @param[in,out] valbuf: Buffer into which the next key's value will be copied.
+ * @param valbuf_sz: Size of @p valbuf
+ * @param[out] val_len: Length of @p val.
+ * @param[out] eof: If true, no more key-value pairs in sequence.
+ *
+ * @remark @p cursor must not be NULL.
+ * @remark @p key must not be NULL.
+ * @remark @p key_len must not be NULL.
+ * @remark @p val must not be NULL.
+ * @remark @p val_len must not be NULL.
```

Comment:
I don't think this remark is valid?

Reply:
You are correct.  I will edit.

Reply:
Or not.  Will revisit making ptrs optional.